be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -2692,13 +2715,16 @@ void SegmentIterator::_calculate_pred_in_remaining_conjunct_root(
 
     auto node_type = expr->node_type();
     if (node_type == TExprNodeType::SLOT_REF) {
+        auto slot_expr = std::dynamic_pointer_cast<doris::vectorized::VSlotRef>(expr);
```

Comment:
need if (slot_expr != nullptr) ?

Reply:
if (node_type == TExprNodeType::SLOT_REF)  ， it must be doris::vectorized::VSlotRef